src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2283,7 +2283,12 @@ void UpdateMemoryCost(in UInt256 position, in UInt256 length)
                             break;
                         }
 
-                        Span<byte> initCode = vmState.Memory.LoadSpan(in memoryPositionOfInitCode, initCodeLength);
+                        if (!vmState.Memory.MemCodeCache.TryGet((memoryPositionOfInitCode, initCodeLength), out CodeInfo? codeInfo))
+                        {
+                            Span<byte> initCode = vmState.Memory.LoadSpan(in memoryPositionOfInitCode, initCodeLength);
+                            codeInfo = new CodeInfo(initCode.ToArray());
+                            vmState.Memory.MemCodeCache.Set((memoryPositionOfInitCode, initCodeLength), codeInfo);
```

Comment:
Somewhat dislike that cache should be owned by memory, but its used directly here.